### PR TITLE
Numpydocify FontManager.findfont()

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Texts are mainly taken from the original docstring, but some of them are not quite clear yet. See below.

If somebody wants to immediately enhance the docstring, direct pushes are welcome.

### CODE REVIEW

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -1177,25 +1177,39 @@ def score_size(self, size1, size2):
...
-        returned.  If no matches below a certain threshold are found,
-        the default font (usually DejaVu Sans) is returned.
-
-        `directory`, is specified, will only return fonts from the
-        given directory (or subdirectory of that directory).
+        Find a font that most closely matches the given font properties.
+
+        Parameters
+        ----------
+        prop : `~matplotlib.fontmanager.FontProperties`
```
Comment (timhoffm) [inline lib/matplotlib/font_manager.py]: Apparently, this supports strings as well. I can do `findfont('Calibri')` or `findfont('Calibri:bold:')` what is the correct terminology for the string representation?

File: lib/matplotlib/font_manager.py [hunk 2]
```diff
@@ -1177,25 +1177,39 @@ def score_size(self, size1, size2):
...
+            The font properties to search for.
+        fontext : str, optional, default: 'ttf'
```
Comment (timhoffm) [inline lib/matplotlib/font_manager.py]: Do we support extensions other than 'ttf' and 'afm'?

The code says:
~~~
        if fontext == 'afm':
            fontlist = self.afmlist
        else:
            fontlist = self.ttflist
~~~

Not sure if extensions other than 'ttf' can appear in `self.ttflist`.

File: lib/matplotlib/font_manager.py [hunk 3]
```diff
@@ -1177,25 +1177,39 @@ def score_size(self, size1, size2):
...
+            The extension of font file; e.g. 'ttf' or 'afm'.
+        directory : str, optional
+            If given, only search this directory and its subdirectories.
+        fallback_to_default : bool
+            If True, will fallback to the default font family (usually
+            "DejaVu Sans" or "Helvetica") if the first lookup hard-fails.
```
Comment (timhoffm) [inline lib/matplotlib/font_manager.py]: I don't quite understand the behavior for `fallback_to_default=False`. This also returns "DejaVu Sans". Not sure if this is a bug. Anyway, we're missing documentation for the non-fallback result.

File: lib/matplotlib/font_manager.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: it's a "fontconfig pattern" (mentioned elsewhere in the module).

File: lib/matplotlib/font_manager.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: Really it's "ttf"->TrueType & OpenType (.ttf, .ttc, .otf) and "afm"->Adobe Font Metrics (.afm) (see get_fontext_synonyms).

File: lib/matplotlib/font_manager.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: In one case we still try to match the font size, weight, and style (italic, bold, etc.) but not in the other.  TBH I doubt the distinction really matters and would suggest just deprecating that argument (can be done in a separate PR, of course).

Comment (timhoffm) [reply lib/matplotlib/font_manager.py]: Ok. I'll take care of that later in a separate PR.

**Pull request merged**